### PR TITLE
Chapter 13: User Microposts

### PR DESCRIPTION
**Pull request by ainulfarizah**
ready

### CODE REVIEW

**Comment by ainulfarizah** [comment]
ready

File: app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: %i(create destroy)
+  before_action :correct_user, only: :%i(destroy)
```
Comment (ghost) [inline app/controllers/microposts_controller.rb]:
```suggestion
  before_action :correct_user, only: %i(destroy)
```

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -8,4 +8,13 @@
       </h1>
     </section>
   </aside>
+  <div class="col-md-8">
+    <% if @microposts.any? %>
+    <h3>Microposts (<%= @microposts.count %>)</h3>
+    <ol class="microposts">
+    <%= render @microposts %>
+    </ol>
+    <%= will_paginate @microposts %>
```
Comment (ghost) [inline app/views/users/show.html.erb]: indent please

File: test/controllers/account_activations_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (ghost) [inline test/controllers/account_activations_controller_test.rb]: please delete this file

File: test/mailers/previews/user_mailer_preview.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Preview all emails at http://localhost:3000/rails/mailers/user_mailer
+class UserMailerPreview < ActionMailer::Preview
```
Comment (ghost) [inline test/mailers/previews/user_mailer_preview.rb]: please delete this file

File: test/mailers/user_mailer_test.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+require "test_helper"
+
+class UserMailerTest < ActionMailer::TestCase
```
Comment (ghost) [inline test/mailers/user_mailer_test.rb]: please delete this file

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ainulfarizah** [comment]
ready

File: app/views/user_mailer/account_activation.html.erb [hunk 6]
```diff
@@ -4,3 +4,4 @@
 <p> <%= t(:link_activate_account) %> </p>
 
 <%= link_to t(:activate), edit_account_activation_url(id: @user.activation_token, email: @user.email) %>
+
```
Comment (ngocvt-0484) [inline app/views/user_mailer/account_activation.html.erb]: remove blank line

File: config/locales/en.yml [hunk 7]
```diff
@@ -1,4 +1,4 @@
-en:
+-en:
```
Comment (ngocvt-0484) [inline config/locales/en.yml]:
```suggestion
en:
```

**Comment by ainulfarizah** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**